com.wso2telco.dep.common.mediation/src/main/java/com/wso2telco/dep/common/mediation/MSISDNBlacklistWhitelistMediator.java

```diff
@@ -0,0 +1,115 @@
+                                   String errorText, String errorVariable, String httpStatusCode,
+                                   String exceptionType) {
+
+        synContext.setProperty(ContextPropertyName.MESSAGE_ID, messageId);
+        synContext.setProperty("mediationErrorText", errorText);    // mediationErrorText  ContextPropertyName.ERROR_TEXT
+        synContext.setProperty(ContextPropertyName.ERROR_VARIABLE, errorVariable);
+        synContext.setProperty(ContextPropertyName.HTTP_STATUS_CODE, httpStatusCode);
+        synContext.setProperty(ContextPropertyName.EXCEPTION_TYPE, exceptionType);
+    }
+
+    public boolean mediate(MessageContext messageContext) {
+
+        String msisdn = (String) messageContext.getProperty("paramValue");
+        String paramArray = (String) messageContext.getProperty("paramArray");
+        String maskedMsidsn = (String) messageContext.getProperty("MASKED_MSISDN");
+        String maskedMsisdnSuffix = (String) messageContext.getProperty("MASKED_MSISDN_SUFFIX");
+        String apiName = (String) messageContext.getProperty("API_NAME");
+        String apiVersion = (String) messageContext.getProperty("VERSION");
+        String spName = ((String) messageContext.getProperty("USER_ID")).split("@")[0];
+        String appID = messageContext.getProperty("api.ut.application.id").toString();
+        String apiID = (String) messageContext.getProperty("API_ID");
+        String action = (String) messageContext.getProperty("ACTION");
+
+        String loggingMsisdn = msisdn;
+
+        String regexPattern = (String) messageContext.getProperty("msisdnRegex");
+        String regexGroupNumber = (String) messageContext.getProperty("msisdnRegexGroup");
+
+        Pattern pattern = Pattern.compile(regexPattern);
+        Matcher matcher = pattern.matcher(msisdn);
+
+        String formattedPhoneNumber = null;
+        if(Boolean.parseBoolean((String)messageContext.getProperty("USER_ANONYMIZATION"))) {
+            loggingMsisdn = maskedMsidsn;
+            formattedPhoneNumber = maskedMsisdnSuffix;
+        } else if (matcher.matches()) {
+            formattedPhoneNumber = matcher.group(Integer.parseInt(regexGroupNumber));
+        }
+
+        APIService apiService = new APIService();
+
+        try {
+            if (apiService.isBlackListedorWhiteListedNumber(formattedPhoneNumber,apiID,appID,spName,action)) {
+                log.info(loggingMsisdn + " is action:" + action +" number for " + apiName + " API " + apiVersion + " version");
+
+                if(action.equalsIgnoreCase(MSISDNConstants.BLACKLIST)){
+                    messageContext.setProperty(SynapseConstants.ERROR_CODE, "POL0001:");
+                    messageContext.setProperty(SynapseConstants.ERROR_MESSAGE, "Internal Server Error. action: " +action+
+                            "Number");
+                    messageContext.setProperty(MSISDNConstants.BLACKLIST, "true");
```

Comment:
messageContext.setProperty assigning values in line number 65, 66, 67, and 68 are duplicates in line number 81, 82, 83 and 84 (apart from the . Will have a separate common method to update these with the MSISDNConstants and boolean passed as parameters (for line number 68 and 84 values)

Finally, you will have a commonplace to update the messageContext in both the scenarios

Reply:
Done